chromium_src/components/omnibox/browser/autocomplete_controller.cc

```diff
@@ -3,13 +3,25 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "brave/components/brave_search_conversion/utils.h"
+#include "brave/components/omnibox/browser/promotion_provider.h"
+#include "brave/components/omnibox/browser/promotion_utils.h"
 #include "brave/components/omnibox/browser/suggested_sites_provider.h"
 #include "brave/components/omnibox/browser/topsites_provider.h"
 
-#define BRAVE_AUTOCOMPLETE_CONTROLLER_AUTOCOMPLETE_CONTROLLER         \
-  providers_.push_back(new TopSitesProvider(provider_client_.get())); \
-  providers_.push_back(new SuggestedSitesProvider(provider_client_.get()));
+#define BRAVE_AUTOCOMPLETE_CONTROLLER_AUTOCOMPLETE_CONTROLLER               \
+  providers_.push_back(new TopSitesProvider(provider_client_.get()));       \
+  providers_.push_back(new SuggestedSitesProvider(provider_client_.get())); \
+  if (!provider_client_->IsOffTheRecord())                                  \
+    providers_.push_back(new PromotionProvider(provider_client_.get()));
```

Comment:
@simonhong I think one more thing that we can add into this PR is to skip this and other `Init`-like places when both features are turned off to make things safe in case we missed something.
you can make a bool function like `bool IsBraveSearchConversionEnabled()` and insert it into some places to protect ourselves.